src/Engine/MapController.java

```diff
@@ -622,6 +628,31 @@ private void startNextTurn()
 
     // Kick off the animation cycle, which will animate/init each unit.
     changeInputMode(InputMode.ANIMATION);
+
+    // Here are the fog-drawing rules. If there are:
+    //   zero humans - spectating - draw everything the current player sees.
+    //   one human - player vs ai - draw everything the human player could see.
+    //   2+ humans - player vs player - draw what the current player sees, IFF the player is human.
+    if( !myGame.activeCO.isAI() )
```

Comment:
You changed the logic here from what it was before - with no humans in play, `myView.gameMap` isn't set at all.

I'm also just generally uncomfortable with having `MapController` in charge of the `GameMap` used by `SpriteMapView` - with the exception of dumb struct-like classes, I like classes to be in control of their own members. I'd also prefer the UI code to be the one looking at the game-logic code, and the game code to have as little idea as possible that the UI exists.